concurrent/src/main/java/io/airlift/concurrent/MoreFutures.java

```diff
@@ -494,6 +495,11 @@ public void onFailure(Throwable t)
     }
 
     public static <T> void addSuccessCallback(ListenableFuture<T> future, Consumer<T> successCallback)
```

Comment:
It would be nice to add some javadoc here that mentions that direct executor will be used here, and the consequences of that.